src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected readonly PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            if (pyObj == null)
+                throw new PythonException();
+            pyObject = pyObj;
+        }
+
+        private void propagateIterationException()
+        {
+            var err = Runtime.PyErr_Occurred();
+            if (err == null) return;
+
+            //remove StopIteration exceptions
+            if (0 != Runtime.PyErr_ExceptionMatches(Exceptions.StopIteration))
+            {
+                Runtime.PyErr_Clear();
+                return;
+            }
+
+            Runtime.CheckExceptionOccurred();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
```

Comment:
Because `IEnumerator<T>` also implements `IEnumerator`, you don't actually need this method to have full body. You can just replace it with `IEnumerator IEnumerable.GetEnumerator() => this.GetEnumerator();`

Reply:
@lostmsu thanks.  I was trying to remember what the shorthand for this was when I was implementing it but couldn't and just duplicated it.  I knew it was something this simple..